pkg/piecestore/pstore.go

```diff
@@ -32,73 +32,85 @@ func PathByHash(hash, dir string) (string, error) {
 	return path.Join(dir, folder1, folder2, fileName), nil
 }
 
-// Store -- Store data into piece store
-// 	hash 					(string)		Hash of the data to be stored
-// 	r 			  		(io.Reader)	File/Stream that contains the contents of the data to be stored
-// 	length 				(length)		Size of the data to be stored
-// 	psFileOffset 	(offset)  	Offset of the data that you are writing. Useful for multiple connections to split the data transfer
-// 	dir 					(string)		pstore directory containing all other data stored
-// 	returns 			(error) 		error if failed and nil if successful
-func Store(hash string, r io.Reader, length int64, psFileOffset int64, dir string) (int64, error) {
+// StoreWriter -- Store data into piece store in multiple writes
+// 	hash 					(string)								Hash of the data to be stored
+// 	dir 					(string)								pstore directory containing all other data stored
+// 	returns 			(*fpiece.Chunk, error) 	error if failed and nil if successful
+func StoreWriter(hash string, length int64, psFileOffset int64, dir string) (*fpiece.Chunk, error) {
 	if psFileOffset < 0 {
-		return 0, ArgError.New("Offset is less than 0. Must be greater than or equal to 0")
-	}
-	if length < 0 {
-		return 0, ArgError.New("Length is less than 0. Must be greater than or equal to 0")
+		return nil, ArgError.New("Offset is less than 0. Must be greater than or equal to 0")
 	}
 	if dir == "" {
-		return 0, ArgError.New("No path provided")
+		return nil, ArgError.New("No path provided")
```

Comment:
everywhere you use PathByHash you check that dir != "". Just move that check into PathByHash and then you don't need to check if dir != "" everywhere else.

i'm also curious - i've made this very comment a number of times. i don't know if the comments are getting lost or what - if you're getting my code review comments can you reply with an emoji or something just so i know i don't have to keep remaking them?

Reply:
Sometimes I make commits and some of the comments disappear. I'll forget to do something when the comments disappear. I wish I had to manually mark them as resolved instead of them automatically disappearing